impress-org/givewp
# Fix: update billing address dynamic validation rules

Resolves [GIVE-203]

## Description



1. Since the introduction of the `'optional'` validation rule as the default, dynamic rules (like the ones used in the billing address) were not taking affect since they were being skipped when empty.  For these fields specifically, the validation is dynamic - based on other field values.  To solve this here we can just move the `optional` rule last. 

2. The front-end templates needed to be updated pre-render to dynamically display the asterisk.

## Affects



- The billing address field

## Visuals



<img width="731" alt="Screenshot 2024-02-01 at 11 29 13 AM" src="https://github.com/impress-org/givewp/assets/10138447/f9efb7ba-32f1-49b7-80e0-b3deaf12fa45">


## Testing Instructions



Zip: https://github.com/impress-org/givewp/actions/runs/7744622366

- Create a v3 form with billing address block
- Select US and do not fill out zip and try to donate
- It should display a zip is required
- Now test out other countries
- Using list below you should see the zip field required asterisk disappear when selecting these countries


Below are countries that do not require zip:

<img width="1143" alt="Screenshot 2024-02-01 at 11 25 41 AM" src="https://github.com/impress-org/givewp/assets/10138447/2e287ed1-878e-414c-8448-dceceb696ed3">

**Singapore**
- Should not require city


## Pre-review Checklist






[GIVE-203]: https://stellarwp.atlassian.net/browse/GIVE-203?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
